src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -399,6 +399,13 @@ private bool TryGetValue(FieldsToFetch.FieldToFetch fieldToFetch, Document docum
                     }
                 }
 
+                else if (fieldToFetch.CanExtractFromIndex)
+                {
+                    var field = luceneDoc.GetField(fieldToFetch.QueryField.SourceAlias);
```

Comment:
Maybe check it `field == null` and throw exception then (unless we are guaranteed that the field exists at this point)

Reply:
yes we know we that the field exists at this point

Reply:
@aviv86 ping

Reply:
Actually the field can be missing if the index has dynamic fields. fixing that